pkg/autodiscovery/providers/kube_endpoints.go

```diff
@@ -144,11 +144,13 @@ func parseServiceAnnotationsForEndpoints(services []*v1.Service) []configInfo {
 			log.Debug("Ignoring a nil service")
 			continue
 		}
-		endptConf, errors := extractTemplatesFromMap(apiserver.EntityForEndpoints(svc.Namespace, svc.Name, ""), svc.Annotations, kubeEndpointAnnotationPrefix)
+		endpointsID := apiserver.EntityForEndpoints(svc.Namespace, svc.Name, "")
+		endptConf, errors := extractTemplatesFromMap(endpointsID, svc.Annotations, kubeEndpointAnnotationPrefix)
 		for _, err := range errors {
 			log.Errorf("Cannot parse endpoint template for service %s/%s: %s", svc.Namespace, svc.Name, err)
 		}
 		for i := range endptConf {
+			endptConf[i].Source = "kube_endpoints:" + endpointsID
```

Comment:
Can we modify the method `extractTemplatesFromMap ` to set directly the value on `.Source`

Reply:
`extractTemplatesFromMap` is a generic util used by `kube_services` and `docker`config providers as well

Reply:
we can add another parameter to the method that provide the prefix `kube_endpoints:` or `kube_service:`...